src/liballoc/vec.rs

```diff
@@ -2340,18 +2340,44 @@ impl<T> Iterator for IntoIter<T> {
 
     #[inline]
     fn size_hint(&self) -> (usize, Option<usize>) {
-        let exact = if mem::size_of::<T>() == 0 {
-            (self.end as usize).wrapping_sub(self.ptr as usize)
-        } else {
-            unsafe { self.end.offset_from(self.ptr) as usize }
-        };
+        let exact = self.len();
         (exact, Some(exact))
     }
 
     #[inline]
     fn count(self) -> usize {
         self.len()
     }
+
+    #[inline]
+    fn nth(&mut self, n: usize) -> Option<T> {
+        unsafe {
+            if self.ptr as *const _ == self.end {
+                None
+            } else {
+                let new_ptr = if mem::size_of::<T>() == 0 {
+                    // purposefully don't use 'ptr.offset' because for
+                    // vectors with 0-size elements this would return the
+                    // same pointer.
+                    arith_offset(self.ptr as *const i8, n as isize) as *mut T
+                } else {
+                    self.ptr.offset(n as isize)
```

Comment:
Offsetting (more than one) past the end is [undefined behaviour](https://doc.rust-lang.org/std/primitive.pointer.html#safety-1). Even this was `wrapping_offset`, this would still be broken when wrapping around the address space.

Reply:
Yeah.  And even so this will leak all the elements we jump over.  I'm just going to remove `nth`.